ambry-router/src/test/java/com.github.ambry.router/OperationTrackerTest.java

```diff
@@ -152,6 +159,133 @@ public void localFailTest() {
     assertTrue("Operation should be done", ot.isDone());
   }
 
+  /**
+   * Test put operation in local dc when replicasStateEnabled is enabled/disabled.
+   * Test steps:
+   * Case1: Only 2 STANDBY replicas in local dc;
+   *        Make 1 succeed and the other fail (both current and replicaState enabled tracker should fail)
+   * Case2: 2 STANDBY, 1 INACTIVE replicas in local dc
+   *        Make 1 fail and 2 succeed (replicaState enabled operation tracker should fail)
+   * Case3: 1 LEADER, 4 STANDBY and 1 INACTIVE in local dc
+   *        Make 3 succeed and 2 fail (replicaState enabled operation tracker should fail)
+   */
+  @Test
+  public void localPutWithReplicaStateTest() {
+    assumeTrue(operationTrackerType.equals(SIMPLE_OP_TRACKER));
+    List<Port> portList = Collections.singletonList(new Port(PORT, PortType.PLAINTEXT));
+    List<String> mountPaths = Collections.singletonList("mockMountPath");
+    datanodes = Collections.singletonList(new MockDataNodeId(portList, mountPaths, "dc-0"));
+    mockPartition = new MockPartitionId();
+    // test that if there are only 2 eligible replicas, the success target should use routerConfig.routerPutSuccessTarget
+    populateReplicaList(2, ReplicaState.STANDBY);
+    localDcName = datanodes.get(0).getDatacenterName();
+    mockClusterMap = new MockClusterMap(false, datanodes, 1, Collections.singletonList(mockPartition), localDcName);
+    OperationTracker ot = getOperationTracker(true, 1, 2, true, Integer.MAX_VALUE, RouterOperation.PutOperation);
+    assertFalse("Operation should not have been done.", ot.isDone());
+    sendRequests(ot, 2, false);
+    // make one requests succeed, the other fail
+    ot.onResponse(inflightReplicas.poll(), TrackedRequestFinalState.SUCCESS);
+    ot.onResponse(inflightReplicas.poll(), TrackedRequestFinalState.FAILURE);
+    assertFalse("Operation should fail", ot.hasSucceeded());
```

Comment:
the successTarget is 1 here, why would it fail?

Reply:
successTarget is 2 for PUT operation. This test is for PUT scenario.